src/generator/audit.jl

```diff
@@ -1,20 +1,41 @@
+function _valid_lib_name(str)
+    try
+        Meta.parse(str)
+        return true
+    catch err
+        return false
+    end
+end
+
 """
     audit_library_name(dag::ExprDAG, options::Dict)
 Check whether the library name is valid.
 """
-function audit_library_name(dag::ExprDAG, options::Dict)
+function audit_library_name(trans_units::Vector{TranslationUnit}, dag::ExprDAG, options::Dict)
     general_options = get(options, "general", Dict())
+    headers = [basename(tu.source_filename) for tu in trans_units]
+    library_names = get(general_options, "library_names", Dict())
+
+    # Check if `library_name` is valid if it exists. If it doesn't exist, check
+    # that the user set library names in `library_names` for all the headers.
     if haskey(general_options, "library_name")
-        try
-            Meta.parse(general_options["library_name"])
-        catch err
+        if !_valid_lib_name(general_options["library_name"])
             general_options["library_name"] = "libxxx"
             @warn "failed to parse the `library_name` as a valid Julia expression, default library name: \":libxxx\" is being used instead."
         end
-    else
+    elseif !issubset(headers, keys(library_names))
         general_options["library_name"] = "libxxx"
-        @warn "default libname: \":libxxx\" is being used, did you forget to set `library_name` in the toml file?"
+        @warn "default libname: \":libxxx\" is being used because `library_names` is missing entries, did you forget to set `library_name` in the toml file?"
```

Comment:
this warning is a bit misleading. it sounds like `library_names` needs to be an exhaustive list, but actually not necessary to be.

to keep it simple and stupid, maybe it's a good idea to let users always define `library_name` to suppress the warning.

Reply:
to change the old warning to something like:

```
@warn "default libname: \":libxxx\" is being used, did you forget to set `library_name` in the toml file? It's safe to ignore this warning if you are using `library_names` as an exhaustive list."
```

Reply:
Ok, I changed it in af62eff.